client/rpc/txpool/src/lib.rs

```diff
@@ -0,0 +1,154 @@
+// (at your option) any later version.
+
+// Moonbeam is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Moonbeam.  If not, see <http://www.gnu.org/licenses/>.
+
+use ethereum::{Transaction as EthereumTransaction, TransactionMessage};
+use ethereum_types::{H160, H256, U256};
+use jsonrpc_core::Result as RpcResult;
+use jsonrpc_core::{Error, ErrorCode};
+pub use moonbeam_rpc_core_txpool::{
+	GetT, Summary, Transaction, TransactionMap, TxPool as TxPoolT, TxPoolResult, TxPoolServer,
+};
+use sha3::{Digest, Keccak256};
+use sp_api::{BlockId, ProvideRuntimeApi};
+use sp_blockchain::{Error as BlockChainError, HeaderBackend, HeaderMetadata};
+use sp_runtime::traits::Block as BlockT;
+use sp_transaction_pool::{InPoolTransaction, TransactionPool};
+use std::collections::HashMap;
+use std::{marker::PhantomData, sync::Arc};
+
+use moonbeam_rpc_primitives_txpool::TxPoolRuntimeApi;
+
+// Note: both `internal_err` and `public_key` could be imported from Frontier.
+// However there are dependency collisions due to Frontier using a different Substrate branch.
+// This may change in the future, and in that case we could just import them.
+pub fn internal_err<T: ToString>(message: T) -> Error {
+	Error {
+		code: ErrorCode::InternalError,
+		message: message.to_string(),
+		data: None,
+	}
+}
+
+pub fn public_key(
+	transaction: &EthereumTransaction,
+	hash: H256,
+) -> Result<[u8; 64], sp_io::EcdsaVerifyError> {
+	let mut sig = [0u8; 65];
+	let mut msg = [0u8; 32];
+	sig[0..32].copy_from_slice(&transaction.signature.r()[..]);
+	sig[32..64].copy_from_slice(&transaction.signature.s()[..]);
+	sig[64] = transaction.signature.standard_v();
+	msg.copy_from_slice(&hash[..]);
+
+	sp_io::crypto::secp256k1_ecdsa_recover(&sig, &msg)
+}
+
+pub struct TxPool<B: BlockT, C, P> {
+	client: Arc<C>,
+	pool: Arc<P>,
+	_marker: PhantomData<B>,
+}
+
+impl<B, C, P> TxPool<B, C, P>
+where
+	C: ProvideRuntimeApi<B>,
+	C: HeaderMetadata<B, Error = BlockChainError> + HeaderBackend<B> + 'static,
+	C: Send + Sync + 'static,
+	B: BlockT<Hash = H256> + Send + Sync + 'static,
+	P: TransactionPool<Block = B> + Send + Sync + 'static,
+	C::Api: TxPoolRuntimeApi<B>,
+{
+	fn map_build<T>(&self) -> RpcResult<TransactionMap<T>>
+	where
+		T: GetT,
+	{
+		let txs: Vec<<B as BlockT>::Extrinsic> = self
+			.pool
+			.ready()
+			.map(|in_pool_tx| in_pool_tx.data().clone())
+			.collect();
+
+		let best_block: BlockId<B> = BlockId::Hash(self.client.info().best_hash);
+		let ethereum_txns = self
+			.client
+			.runtime_api()
+			.extrinsic_filter(&best_block, txs)
+			.map_err(|err| {
+				internal_err(format!("fetch runtime extrinsic filter failed: {:?}", err))
+			})?;
+		let mut out: TransactionMap<T> = HashMap::new();
```

Comment:
```suggestion
		let mut out = TransactionMap::<T>::new();
```
`TransactionMap` is a type alias, so you can directly use its functions

Reply:
Cheers!